src/Common/HyperLogLogCounter.h

```diff
@@ -264,7 +264,8 @@ enum class HyperLogLogMode
 /// of Algorithms).
 template <
     UInt8 precision,
-    typename Hash = IntHash32<UInt64>,
+    typename Key = UInt64,
```

Comment:
I dont' get the logic behind the changes in this file.

Reply:
There is no way to get type of `Key` for hashing, I've tried to do some meta programming, but failed because operator() cannot be static. So that's why separate template argument had been introduced instead, to cast using this type.

Reply:
Alright, thanks.